lib/server.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+
+const net = require('net');
+const EventEmitter = require('events');
+const Connection = require('./connection');
+const Watcher = require('./watcher');
+const { isDir } = require('./fs-interface');
+
+const emitRemoteEvent = Symbol('emitRemoteEvent');
+const onSync = Symbol('onSync');
+const onInspect = Symbol('onInspect');
+
+const DEFAULT_OPTIONS = {
+  dir: './',
+  host: 'localhost',
+  port: 8080,
+};
+
+class Server extends EventEmitter {
+  constructor(options = DEFAULT_OPTIONS) {
+    super();
+
+    const { dir, ignore, port, host } = options;
+    this.dir = dir || DEFAULT_OPTIONS.dir;
+    this.host = host || DEFAULT_OPTIONS.host;
+    this.port = port || DEFAULT_OPTIONS.port;
+    this.connections = new Map();
+    this.connectionId = 1;
+    this.server = new net.Server();
+    this.watcher = new Watcher(this.dir, { ignore });
+
+    this.start = this.start.bind(this);
+    this.stop = this.stop.bind(this);
```

Comment:
unnecessary reassignment

Reply:
An insurance of `this` existing, just in case of incorrect Server usage. Personally, I stumble onto this problem, writing a test, and after adding this additional binding, Server can handle both cases: correct and slightly wrong, where `this` is `undefined`